src/other-scripts/corrections-modal/index.js

```diff
@@ -0,0 +1,267 @@
+/**
+ * WordPress dependencies.
+ */
+import { useState, useEffect } from '@wordpress/element';
+import { select } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { Button, Modal, PanelBody, TextareaControl, SelectControl, Popover, DateTimePicker } from '@wordpress/components';
+import { registerPlugin } from '@wordpress/plugins';
+import { PluginDocumentSettingPanel } from '@wordpress/editor';
+import { Icon, trash, create } from '@wordpress/icons';
+
+/**
+ * Internal dependencies.
+ */
+import './style.scss';
+import moment from 'moment';
+
+/**
+ * Correction types.
+ *
+ * @type {Object[]} The correction types.
+ */
+const types = [
+	{ label: __( 'Correction', 'newspack-plugin' ), value: 'correction' },
+	{ label: __( 'Clarification', 'newspack-plugin' ), value: 'clarification' },
+];
+
+/**
+ * Save the corrections data.
+ *
+ * @param {number} postId  The post ID.
+ * @param {Object} payload The corrections data.
+ *
+ * @return {Promise} The fetch promise.
+ */
+const saveData = async ( postId, payload ) => {
+	try {
+		const response = await fetch( `${window.NewspackCorrectionsData.restUrl}/${postId}`, {
+			method: 'POST',
+			headers: {
+				'Content-Type': 'application/json',
+				'X-WP-Nonce': window.NewspackCorrectionsData.nonce,
+			},
+			body: JSON.stringify( payload ),
+		} );
+		return await response.json();
+	} catch ( error ) {
+		throw new Error( error );
+	}
+};
+
+/**
+ * The corrections modal component.
+ *
+ * @return {JSX.Element} The corrections modal component.
+ */
+const CorrectionsModal = () => {
+	/**
+	 * Get the current post ID.
+	 */
+	const postId = select('core/editor').getCurrentPostId();
```

Comment:
In a component we should use the `useSelect( select => select( 'core/editor' ).getCurrentPostId() )` hook instead of calling it directly.